tests/functional/test_calculator.py

```diff
@@ -46,12 +49,39 @@ def test_calculator(self):
 
         self.assert_feature_success('features/calculator.feature')
 
-    def test_wrong_expectations(self):
+    def test_failure(self):
         """
         Test that a failing feature fails tests.
         """
 
-        self.assert_feature_fail('features/wrong_expectations.feature')
+        stream = str_io()
+
+        self.assert_feature_fail('features/wrong_expectations.feature',
+                                 stream=stream)
+
+        # Check that the appropriate error messages were printed
+
+        output = stream.getvalue()
+
+        error_header = "FAIL: Add two numbers " + \
+            "(features.wrong_expectations: Wrong expectations)"
+
+        self.assertIn(error_header, output)
+
+        feature_stack_frame = """
+  File "features/wrong_expectations.feature", line 10, in Add two numbers
+    Then the result should be 40 on the screen
+        """.strip()
```

Comment:
It might be worth adding a comment to wrong_expectations.feature to say the line numbers matter. Also I suppose to highlighting.feature.